packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -46,7 +50,10 @@ function SelectionTool(props: Props) {
   const { worldToData } = useVisCanvasContext();
 
   const startEvtRef = useRef<CanvasEvent<PointerEvent>>();
+  const hasSuccessfullyEndedRef = useRef<boolean>(false);
```

Comment:
This ref allows me to track whether the modifier keys where indeed correctly pressed when the `pointerup` event occurred, and also to indicate whether the selection ended "successfully" - i.e. that it didn't get cancelled with <kbd>Escape</kbd>.